File: app/backend/app/schemas/city.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class CityCreate(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    slug: str = Field(min_length=1, max_length=80)
    timezone: str = Field(default="America/Chicago", min_length=1, max_length=80)


class CityPatch(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=80)
    slug: str | None = Field(default=None, min_length=1, max_length=80)
    timezone: str | None = Field(default=None, min_length=1, max_length=80)
    is_active: bool | None = None


class CityOut(BaseModel):
    id: str
    name: str
    slug: str
    timezone: str
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: app/frontend/src/types/city.ts
export interface City {
  id: string
  name: string
  slug: string
  timezone: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export type CityCreate = {
  name: string
  slug: string
  timezone: string
}

export type CityPatch = Partial<CityCreate> & {
  is_active?: boolean
}
